### PR TITLE
[Reference Model] Chapter 3 - Clarify design principle: Declarative

### PR DESCRIPTION
**Pull request by BernardTsai-DT**
#201 Clarify the intention of the design principle: Declarative

### CODE REVIEW

**Comment by pgoyal01** [comment]
@BernardTsai-DT The intent, in my opinion, is the same as "Infrastructure as Code" (IAC).  As you know, the IAC concept allow infrastructure to be treated the same way as application programs -- human readable code specification.  The IAC code allows us to declaratively script the final state of the infrastructure environment; the execution of the IAC code automatically provisions the target environment. The software development process including change management are applicable and hence can adapt the CI/CD framework for infrastructure build, deployment, provisioning, updates, etc.

**Comment by BernardTsai-DT** [comment]
@pgoyal01 I totally agree - would you like to modify the statement or does it already capture the view you have expressed in your comment?

**Comment by pgoyal01** [comment]
@BernardTsai-DT Your definition restricts it to (1) machine readable and (2) on-boarding and maintaining workloads. We can have human readable but yet not be declarative (what but not the how).  Does the following work: "The model should allow for the description of the final state of the NFVI resources for automated life cycle management."

**Comment by BernardTsai-DT** [comment]
I very much like the proposal and have integrated the proposal into chapter 3 design principles.

File: doc/ref_model/chapters/chapter03.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ Although a couple of explicit and implicit abstraction models (e.g. in the conte
 -	**Scope:** the model should describe the most relevant virtualised infrastructure resources (incl. acceleration technologies) an NFVI needs to provide for hosting Telco workloads
 -	**Separation of Concern:** the model should support a clear distinction between the responsibilities related to maintaining the network function virtualisation infrastructure and the responsibilities related to managing the various VNF workloads
 -	**Simplicity:** the amount of different types of resources (including their attributes and relationships amongst one another) should be kept to a minimum to reduce the configuration spectrum which needs to be considered
--	**Declarative:** the model should allow for a declarative description of the required NFVI capabilities for on-boarding and maintaining workloads
+- **Declarative**: the model should allow for the description of the final state of the NFVI resources for automated life cycle management
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter03.md]: It is not only the NFVI resources, but also the NFVI configuration what we would need to describe in a declarative way.

Comment (BernardTsai-DT) [reply doc/ref_model/chapters/chapter03.md]: That is correct - I have included the configuration aspect into this principle

**Pull request merged**